From b95b67c75f23fdb2680eabe8f8cc3e0e162552cb Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Fri, 21 Sep 2012 19:59:44 +0200
Subject: [PATCH 42/82] Set correct filter flags.

UATH_FILTER_RX_BEACON was not properly set causing mac to detect beacon
loss all the time and try to probe the AP.
---
 ar5523.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

--- a/drivers/net/wireless/ath/ar5523/ar5523.c
+++ b/drivers/net/wireless/ath/ar5523/ar5523.c
@@ -1079,7 +1079,7 @@ static int ar5523_start(struct ieee80211
 	ar5523_set_rxfilter(ar, 0, UATH_FILTER_OP_INIT);
 	ar5523_set_rxfilter(ar,
 	    UATH_FILTER_RX_UCAST | UATH_FILTER_RX_MCAST |
-	    UATH_FILTER_RX_BCAST,
+	    UATH_FILTER_RX_BCAST | UATH_FILTER_RX_BEACON,
 	    UATH_FILTER_OP_SET);
 
 	ar5523_set_ledsteady(ar, UATH_LED_ACTIVITY, UATH_LED_ON);
@@ -1373,7 +1373,6 @@ out_unlock:
 #define AR5523_SUPPORTED_FILTERS (FIF_PROMISC_IN_BSS | \
 				  FIF_ALLMULTI | \
 				  FIF_FCSFAIL | \
-				  FIF_BCN_PRBRESP_PROMISC | \
 				  FIF_OTHER_BSS)
 
 static void ar5523_configure_filter(struct ieee80211_hw *hw,
@@ -1388,15 +1387,12 @@ static void ar5523_configure_filter(stru
 
 	*total_flags &= AR5523_SUPPORTED_FILTERS;
 
-	if (*total_flags & FIF_BCN_PRBRESP_PROMISC)
-		filter |= UATH_FILTER_RX_BEACON;
-
 	if (*total_flags & FIF_PROMISC_IN_BSS ||
 	    *total_flags & FIF_OTHER_BSS)
 		filter |= UATH_FILTER_RX_PROM;
 
 	filter |= UATH_FILTER_RX_UCAST | UATH_FILTER_RX_MCAST |
-		  UATH_FILTER_RX_BCAST;
+		  UATH_FILTER_RX_BCAST | UATH_FILTER_RX_BEACON;
 
 	ar5523_set_rxfilter(ar, 0, UATH_FILTER_OP_INIT);
 	ar5523_set_rxfilter(ar, filter, UATH_FILTER_OP_SET);
